baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StringBuilderConstantParameters.java

```diff
@@ -95,6 +95,8 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
         return buildDescription(tree)
                 .setMessage(MESSAGE)
                 .addFix(SuggestedFix.builder()
+                        .prefixWith(tree, "(")
+                        .postfixWith(tree, ")")
```

Comment:
I don't think it's desirable to unconditionally wrap the suggested fix with parenthesis. This will result in unidiomatic code in the common case.

Reply:
Do you have any suggestions on how to check if the parens are necessary?

Reply:
Sorry, I don't have any suggestions. I suspect this is fairly non-trivial.

I'm not sure it's worthwhile to fix this issue - I imagine it would significantly complicated the logic here.

Reply:
Yeah, it seems pretty hard.
That said it seems to me that unidiomatic code is preferable to incorrect code- which is what the fixer currently suggests, especially since [UnnecessaryParentheses](https://errorprone.info/bugpattern/UnnecessaryParentheses) will automatically catch and fix the unidiomatic generated code.